pym/portage/package/ebuild/doebuild.py

```diff
@@ -518,6 +526,23 @@ def doebuild_environment(myebuild, mydo, myroot=None, settings=None,
 				mysettings["KV"] = ""
 			mysettings.backup_changes("KV")
 
+	binpkg_compression = mysettings.get("BINPKG_COMPRESSION", "bzip2")
+	try:
+		compression = _compressors[binpkg_compression]
+	except KeyError as e:
+		raise TypeError("Invalid or unsupported compression method: %s" % e.args[0])
```

Comment:
We want emerge to bail out somewhere in  pym/_emerge/actions.py, either in the run_action or action_build function. Actions like emerge --info and emerge --sync should still succeed when the binpkg compression configuration is invalid.